test-suite/src/test/scala/org/scalajs/testsuite/jsinterop/TupleTest.scala

```diff
@@ -0,0 +1,73 @@
+/*                     __                                               *\
+**     ________ ___   / /  ___      __ ____  Scala.js Test Suite        **
+**    / __/ __// _ | / /  / _ | __ / // __/  (c) 2013, LAMP/EPFL        **
+**  __\ \/ /__/ __ |/ /__/ __ |/_// /_\ \    http://scala-js.org/       **
+** /____/\___/_/ |_/____/_/ | |__/ /____/                               **
+**                          |/____/                                     **
+\*                                                                      */
+package org.scalajs.testsuite.jsinterop
+
+import org.scalajs.jasminetest.JasmineTest
+
+import scala.scalajs.js
+
+object TupleTest extends JasmineTest {
+
+  describe("scala.scalajs.js.Tuple") {
+
+    it("should provide an equivalent of Scala tuple") {
+      val obj = js.Tuple2(42, "foobar")
+
+      expect(obj._1).toEqual(42)
+      expect(obj._2).toEqual("foobar")
+    }
+
+    it("should unapply JS tuple in destructuring use case") {
+      val obj = js.Tuple2(42, "foobar")
+      val js.Tuple2(t1, t2) = obj
+
+      val t1IsInt: Int = t1
+      val t2IsString: String = t2
+      expect(t1IsInt).toEqual(42)
+      expect(t2IsString).toEqual("foobar")
+    }
+
+    it("should unapply JS tuple in pattern matching position") {
+      val obj = js.Tuple2(42, "foobar")
+      obj match {
+        case js.Tuple2(2, _) => fail("Not expected match")
+        case js.Tuple2(_ , "foobar") => //ok
```

Comment:
Although here too, you should extract `t1` and `t2` and assert that their types are correctly inferred using `t1IsInt` and `t2IsString`.